Kube-inject panic when using revisions

### Bug Description

I Get a panic when I try to manual sidecar injection and revisions

My kubernetes is simple as I only do the following command

Installed istio 1-13-2 with revision
`istioctl install --set profile=demo -y --revision 1-13-2`

Installed the bookinfo sample application
`kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml`

Finally try to do the manual injection using meshConfigMapName and injectConfigMapName
```
istioctl kube-inject -f samples/bookinfo/platform/kube/bookinfo.yaml --meshConfigMapName istio-1-13-2 --injectConfigMapName istio-sidecar-injector-1-13-2  | kubectl apply -f -

2022-03-23T00:58:42.302388Z	warn	failed to get injection config from mutatingWebhookConfigurations "istio-sidecar-injector", will fall back to get injection from the injection configmap "istio-sidecar-injector" : could not find valid mutatingWebhookConfiguration "sidecar-injector.istio.io" from cluster
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x178 pc=0x102b709b0]

goroutine 1 [running]:
istio.io/istio/pkg/kube/inject.RunTemplate({0x1400077f000, {{0x14000feb540, 0xa}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	istio.io/istio/pkg/kube/inject/inject.go:412 +0x6f0
istio.io/istio/pkg/kube/inject.injectPod({0x1400077f000, {{0x14000feb540, 0xa}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...})
	istio.io/istio/pkg/kube/inject/webhook.go:330 +0xa4
istio.io/istio/pkg/kube/inject.IntoObject({0x103eea1c0, 0x14000fe8c00}, 0x1400080e660, {0x0, 0x0}, {0x0, 0x0}, 0x0, {0x103f28818, 0x140004cd680}, ...)
	istio.io/istio/pkg/kube/inject/inject.go:786 +0xcec
istio.io/istio/pkg/kube/inject.IntoResourceFile({0x103eea1c0, 0x14000fe8c00}, 0x1400080e660, {0x0, 0x0}, {0x0, 0x0}, 0x0, {0x103eeb640, 0x14000113778}, ...)
	istio.io/istio/pkg/kube/inject/inject.go:574 +0x2c0
istio.io/istio/istioctl/cmd.injectCommand.func1(0x14000f6c780, {0x14000b67d40, 0x0, 0x6})
	istio.io/istio/istioctl/cmd/kubeinject.go:594 +0x384
github.com/spf13/cobra.(*Command).execute(0x14000f6c780, {0x14000b67ce0, 0x6, 0x6})
	github.com/spf13/cobra@v1.3.0/command.go:856 +0x668
github.com/spf13/cobra.(*Command).ExecuteC(0x14000f6c500)
	github.com/spf13/cobra@v1.3.0/command.go:974 +0x410
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.3.0/command.go:902
main.main()
	istio.io/istio/istioctl/cmd/istioctl/main.go:39 +0x114
service/details unchanged
serviceaccount/bookinfo-details unchanged```

This definitely worked on previous versions of istio, 1-8-3 was the last time I tested this.
If I am doing it wrong any help would be appreciated but regardless it shouldn't be giving a null pointer exception.

### Version

```prose
$ istioctl version
client version: 1.13.2
control plane version: 1.13.2
data plane version: 1.13.2 (2 proxies)
$ kubectl version --short
Client Version: v1.23.4
Server Version: v1.22.5
```


### Additional Information

After doing a bit of poking around in the kube-inject source code I possibly see the problem and perhaps a solution.

At line 397 it returns nil for meshConfig which is the cause of the nil pointer later
https://github.com/istio/istio/blob/2bcfd34cb20812b988597299191d11cc041e54f6/istioctl/cmd/kubeinject.go#L389-L401

At line 766 proxyConfig gets set to nil and then gets tripped here
https://github.com/istio/istio/blob/2bcfd34cb20812b988597299191d11cc041e54f6/pkg/kube/inject/inject.go#L748-L773
https://github.com/istio/istio/blob/2bcfd34cb20812b988597299191d11cc041e54f6/pkg/kube/inject/inject.go#L409

<h3>Possible Solution</h3>

Interestingly if I comment line 397 out and force it to proceed to get the meshConfig it appears to work fine
https://github.com/istio/istio/blob/2bcfd34cb20812b988597299191d11cc041e54f6/istioctl/cmd/kubeinject.go#L397-L401

Possible alternate fixes could involve patchBytes not returning false during inject. But it's client config seems to always be nil causing the inject to also return nil.
https://github.com/istio/istio/blob/2bcfd34cb20812b988597299191d11cc041e54f6/pkg/kube/inject/inject.go#L731-L748
https://github.com/istio/istio/blob/2bcfd34cb20812b988597299191d11cc041e54f6/istioctl/cmd/kubeinject.go#L74-L78

--- a/istioctl/cmd/kubeinject.go
+++ b/istioctl/cmd/kubeinject.go
@@ -320,6 +320,22 @@ func getInjectConfigFromConfigMap(kubeconfig, revision string) (inject.RawTempla
 			"Use --injectConfigFile or re-run kube-inject with `-i <istioSystemNamespace> and ensure istio-sidecar-injector configmap exists",
 			injectConfigMapName, istioNamespace, err)
 	}
+	// Check if revision matches metadata in configmap
+	configMapRevision := meshConfigMap.ObjectMeta.GetLabels()[label.IoIstioRev.Name]
+	if configMapRevision != revision {
+		// Check if revision is "", if so, let's show "default"
+		providedRevision := revision
+		if providedRevision == "" {
+			providedRevision = "default"
+		}
+		return nil, fmt.Errorf(
+			"revision in injection configmap %q from namespace %q defines revision as %s, but expecting %s. Check injection configmap or use --revision flag.",
+			injectConfigMapName,
+			istioNamespace,
+			configMapRevision,
+			providedRevision,
+		)
+	}
 	// values in the data are strings, while proto might use a
 	// different data type.  therefore, we have to get a value by a
 	// key

--- a/istioctl/cmd/kubeinject.go
+++ b/istioctl/cmd/kubeinject.go
@@ -317,24 +317,8 @@ func getInjectConfigFromConfigMap(kubeconfig, revision string) (inject.RawTempla
 	meshConfigMap, err := client.CoreV1().ConfigMaps(istioNamespace).Get(context.TODO(), injectConfigMapName, metav1.GetOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("could not find valid configmap %q from namespace  %q: %v - "+
-			"Use --injectConfigFile or re-run kube-inject with `-i <istioSystemNamespace> and ensure istio-sidecar-injector configmap exists",
+			"Use --injectConfigFile or re-run kube-inject with `-i <istioSystemNamespace>` and ensure istio-sidecar-injector configmap exists",
 			injectConfigMapName, istioNamespace, err)
-	}
-	// Check if revision matches metadata in configmap
-	configMapRevision := meshConfigMap.ObjectMeta.GetLabels()[label.IoIstioRev.Name]
-	if configMapRevision != revision {
-		// Check if revision is "", if so, let's show "default"
-		providedRevision := revision
-		if providedRevision == "" {
-			providedRevision = "default"
-		}
-		return nil, fmt.Errorf(
-			"revision in injection configmap %q from namespace %q defines revision as %s, but expecting %s. Check injection configmap or use --revision flag.",
-			injectConfigMapName,
-			istioNamespace,
-			configMapRevision,
-			providedRevision,
-		)
 	}
 	// values in the data are strings, while proto might use a
 	// different data type.  therefore, we have to get a value by a
@@ -609,6 +593,9 @@ It's best to do kube-inject when the resource is initially created.
 			injector, meshConfig, err := setupKubeInjectParameters(&sidecarTemplate, &valuesConfig, rev, injectorAddress)
 			if err != nil {
 				return err
+			}
+			if injector.client == nil && meshConfig == nil {
+				return fmt.Errorf("failed to get injection config from mutatingWebhookConfigurations and injection configmap - check injection configmap or pass --revision flag")
 			}
 			var warnings []string
 			templs, err := inject.ParseTemplates(sidecarTemplate)

--- a/istioctl/cmd/kubeinject.go
+++ b/istioctl/cmd/kubeinject.go
@@ -595,7 +595,10 @@ It's best to do kube-inject when the resource is initially created.
 				return err
 			}
 			if injector.client == nil && meshConfig == nil {
-				return fmt.Errorf("failed to get injection config from mutatingWebhookConfigurations and injection configmap - check injection configmap or pass --revision flag")
+				return fmt.Errorf(
+					"failed to get injection config from mutatingWebhookConfigurations and injection configmap - " +
+						"check injection configmap or pass --revision flag",
+				)
 			}
 			var warnings []string
 			templs, err := inject.ParseTemplates(sidecarTemplate)

--- a/releasenotes/notes/38083.yaml
+++ b/releasenotes/notes/38083.yaml
@@ -1,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+issue:
+  - 38083
+releaseNotes:
+  - |
+    **Fixed** nil pointer dereference panic when using kube-inject when not passing a needed revision but also passing injectConfigMapName.

